### PR TITLE
Mlimonci/parent backend

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-udev-device.c [hunk 1]
```diff
@@ -1269,15 +1269,8 @@ fu_udev_device_set_physical_id(FuUdevDevice *self, const gchar *subsystems, GErr
...
-		if (tmp == NULL) {
-			g_set_error_literal(error,
-					    G_IO_ERROR,
-					    G_IO_ERROR_NOT_FOUND,
-					    "failed to find PCI_SLOT_NAME");
-			return FALSE;
-		}
-		physical_id = g_strdup_printf("PCI_SLOT_NAME=%s", tmp);
+		tmp = g_udev_device_get_sysfs_path(udev_device);
+		physical_id = g_strdup(tmp);
```
Comment (hughsie) [inline libfwupdplugin/fu-udev-device.c:1273]: this isn't a physical ID tho -- this is a backend_id -- the physical ID is supposed to be the same for each physical device, even if it has two different sysfs-paths being added by more than and backend. I think this change has the potential to cause regressions IMHO. That said, I think I can see what you're doing -- and maybe we can use the backend ID instead.

Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:1273]: @superm1 I've added a fixup -- can you tell me if that works for you?

Comment (superm1) [reply libfwupdplugin/fu-udev-device.c]: yup it works

File: libfwupdplugin/fu-device.c [hunk 2]
```diff
@@ -1702,6 +1703,86 @@ fu_device_add_parent_physical_id(FuDevice *self, const gchar *physical_id)
...
+ * fu_device_get_parent_backend_ids:
+ * @self: a #FuDevice
+ *
+ * Gets any parent device IDs. If a device is added to the daemon that matches
+ * the physical ID added from fu_device_add_parent_backend_id() then this
+ * device is marked the parent of @self.
+ *
+ * Returns: (transfer none) (element-type utf8) (nullable): a list of IDs
+ *
+ * Since: 1.6.2
```
Comment (superm1) [inline libfwupdplugin/fu-device.c:1716]: this should be 1.9.7

File: libfwupdplugin/fu-device.c [hunk 3]
```diff
@@ -1702,6 +1703,86 @@ fu_device_add_parent_physical_id(FuDevice *self, const gchar *physical_id)
...
+/**
+ * fu_device_has_parent_backend_id:
+ * @self: a #FuDevice
+ * @backend_id: a device physical ID
+ *
+ * Searches the list of parent IDs for a string match.
+ *
+ * Returns: %TRUE if the parent ID exists
+ *
+ * Since: 1.6.2
```
Comment (superm1) [inline libfwupdplugin/fu-device.c:1735]: this should be 1.9.7

File: libfwupdplugin/fu-device.c [hunk 4]
```diff
@@ -1702,6 +1703,86 @@ fu_device_add_parent_physical_id(FuDevice *self, const gchar *physical_id)
...
+ * @self: a #FuDevice
+ * @backend_id: a device physical ID
+ *
+ * Sets any parent device using the physical ID. An parent device is logically
+ * linked to the primary device in some way and can be added before or after @self.
+ *
+ * The IDs are searched in order, and so the order of adding IDs may be
+ * important if more than one parent device might match.
+ *
+ * Since: 1.6.2
```
Comment (superm1) [inline libfwupdplugin/fu-device.c:1766]: this should be 1.9.7

File: libfwupdplugin/fu-udev-device.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c]: Done!

**Review by superm1 — APPROVED** [review]

**Comment by superm1** [comment]
I wonder if we should also look at the representation for DP aux MST devices (synaptics, realtek) too now.  It would be way more logical to set parents for those off the DRM connector they're attached to.

**Comment by hughsie** [comment]
> I wonder if we should also look at the representation for DP aux MST devices (synaptics, realtek) too now.

I'll dig out my realtek_mst device on Monday and see, good idea.

**Pull request merged**